packages/functions/src/dataTriggers.ts

```diff
@@ -311,3 +312,35 @@ export const registerCreatedOrgSecret = functions
       { merge: true }
     );
   });
+
+export const createPublicOrgInfo = functions
+  .region(__functionsZone__)
+  .firestore.document(`${Collection.Organizations}/{organization}`)
+  .onWrite(async (change, context) => {
+    const { organization } = context.params;
+
+    const db = admin.firestore();
+
+    const isDelete = !change.after.exists;
+
+    const publicOrgInfoDocRef = db
+      .collection(Collection.PublicOrgInfo)
+      .doc(organization);
+    const orgData = change.after.data() as OrganizationData;
+
+    if (isDelete) {
+      await publicOrgInfoDocRef.delete();
+      return;
+    } else if (
+      !orgData.displayName ||
+      !orgData.location ||
+      !orgData.emailFrom
```

Comment:
This way only if all of the values are updated, they would be updated to public org...we don't want that:
- cleaner solution, `O(n)`, would be to check all three if they exist and add them to an update object:
```typescript
const updates = ["displayName", "location", "emailFrom"].reduce(
  (acc, curr) => orgData[curr] ? {...acc, [curr]: orgData[curr]} : acc
)
await publicDocRef.set(updates, {merge: true)
```
- a less clean, but slightly more efficient one, `O(1)` (avoiding the looping):
```typescript
await publicOrgInfoDocRef.set({
      displayName: orgData.displayName || "",
      location: orgData.location || "",
      emailFrom: orgData.emailFrom || "",
    })
```
Needless to say, since it's only three fields (`n = 3` in `O(n)` and this is JS and not C rocket science, I would definitely go with the former (cleaner)